assets/js/modules/reader-revenue-manager/datastore/publications.js

```diff
@@ -45,6 +49,94 @@ const baseInitialState = {
 };
 
 const baseActions = {
+	/**
+	 * Syncronizes the onboarding state of the publication with the API.
+	 * Updates the settings on server.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return {void}
+	 */
+	*syncPublicationOnboardingState() {
+		const registry = yield commonActions.getRegistry();
+		const connected = yield commonActions.await(
+			registry
+				.resolveSelect( CORE_MODULES )
+				.isModuleConnected( MODULE_SLUG )
+		);
+
+		// If the module is not connected, do not attempt to sync the onboarding state.
+		if ( ! connected ) {
+			return;
+		}
+
+		yield commonActions.await(
+			registry
+				.resolveSelect( MODULES_READER_REVENUE_MANAGER )
+				.getSettings()
+		);
+
+		const publicationID = registry
+			.select( MODULES_READER_REVENUE_MANAGER )
+			.getPublicationID();
+
+		// If there is no publication ID, do not attempt to sync the onboarding state.
+		if ( publicationID === undefined ) {
+			return;
+		}
+
+		const publications = registry
+			.select( MODULES_READER_REVENUE_MANAGER )
+			.getPublications();
+
+		const publication = publications.find(
+			// eslint-disable-next-line sitekit/acronym-case
+			( { publicationId } ) => publicationId === publicationID
+		);
+
```

Comment:
We should also add a guard here in case the publication is not found:

```suggestion

		// If the publications is not found, do not attempt to sync the onboarding state.
		if ( ! publication ) {
			return;
		}
```